modcc/module.cpp

```diff
@@ -189,6 +189,24 @@ bool Module::semantic() {
     // move functions and procedures to the symbol table
     if(!move_symbols(callables_))  return false;
 
+    // Before starting the inlining process, look for the BREAKPOINT block:
+    // if it includes a SOLVE statement, check that it is the first statement
+    // in the block.
+    if (has_symbol("breakpoint", symbolKind::procedure)) {
+        bool found_non_solve = false;
+        auto breakpoint = symbols_["breakpoint"]->is_procedure();
+        for (const auto& s: breakpoint->body()->statements()) {
+            if(!s->is_solve_statement()) {
+                found_non_solve = true;
+                continue;
+            }
+            if (found_non_solve) {
```

Comment:
should this be `else if (found_non_solve)`, because it is only an error if a SOLVE statement is encountered after the first non-solve statement.

Reply:
Doesn't the `continue` take care of that? 
But I think removing the `continue` and using `else` is clearer, so I'll make the change.

Reply:
Ah, yes, you are right. But I think that the logic is a bit confusing relying on the `continue`.